app/notifiers/notifier/base.rb

```diff
@@ -0,0 +1,79 @@
+# The base class of notifiers. This is meant to be called by the Notifications Framework
+#
+# @api notifications
+class Notifier::Base
+  class ActivityWrapper < SimpleDelegator
+    # Send notifications according to input type and recipient
+    #
+    # @param [Object] recipient The recipient of the notification
+    # @param [Symbol] type The type of notification
+    def notify(recipient, type)
+      super(recipient, type).tap do |notification|
+        case recipient
+        when Course
```

Comment:
why the wrapper? does the wrapper know about the notification or the notifier or both, and why?